dask/array/reductions.py

```diff
@@ -457,7 +457,16 @@ def moment_agg(pairs, order=2, ddof=0, dtype='f8', sum=np.sum, axis=None, **kwar
     inner_term = divide(totals, ns, dtype=dtype) - mu
 
     M = _moment_helper(Ms, ns, inner_term, order, sum, axis, kwargs)
-    return divide(M, n.sum(axis=axis, **kwargs) - ddof, dtype=dtype)
+
+    denominator = n.sum(axis=axis, **kwargs) - ddof
+
+    # taking care of the edge case with empty or all-nans array with ddof > 0
+    if isinstance(denominator, np.ndarray):
```

Comment:
IIUC, this may end up being something like a CuPy array, (not necessarily a NumPy ndarray). Maybe flip the cases of the `if` block and do something like

```python
if isinstance(denominator, numbers.Number):
    denominator = np.nan
else:
    denominator[denominator < 0] = np.nan
```

Reply:
@TomAugspurger thank you, didn't think about that.